pkg/cloudprovider/suite_test.go

```diff
@@ -315,13 +321,17 @@ var _ = Describe("CloudProvider", func() {
 				Expect(drifted).To(Equal(NoDrift))
 			})
 
-			It("should error when node is not found", func() {
+			It("shouldn't error or be drifted when node is not found", func() {
 				nodeClaim.Status.NodeName = "NodeWhoDoesNotExist"
 				drifted, err := cloudProvider.IsDrifted(ctx, nodeClaim)
-				Expect(err).To(HaveOccurred())
+				Expect(err).ToNot(HaveOccurred())
 				Expect(drifted).To(Equal(NoDrift))
 			})
 
+			// TODO (charliedmcb): Do we need to test "shouldn't error or be drifted when node is deleting"?
```

Comment:
You could add a finalizer on the node manually, delete it, run the test, then remove the finalizer if you wanted? It's pretty easy to add a finalizer.

I'm not sure it's strictly needed though because it doesn't seem very likely that your "if deletion timestamp set" has a big, given the simplicity of the statement.

Reply:
Since the case is so simple, I think I'll just leave the TODO comment here, but update it with more of an explanation.

Reply:
Decided to implement the test case.